tools/file_packager.py

```diff
@@ -568,6 +580,17 @@ def was_seen(name):
         with open(options.jsoutput + '.metadata', 'w') as f:
           json.dump(metadata, f, separators=(',', ':'))
 
+  if options.depfile:
+    with open(options.depfile, 'w') as f:
+      for target in [data_target, options.jsoutput]:
+        if target:
+          f.write(target.replace('$', '$$').replace('#', '\\#').replace(' ', '\\ '))
+          f.write(' \\\n')
+      f.write(': \\\n')
+      for dependency in walked:
+        f.write(dependency.replace('$', '$$').replace('#', '\\#').replace(' ', '\\ '))
```

Comment:
Is this the way `make` escapes these chars?  Perhaps create a seprate helper `escape_for_makefille`?

Reply:
I can't find a sufficiently authoritative reference for what Make accepts, but CMake does have a [documented grammar](https://cmake.org/cmake/help/latest/command/add_custom_command.html#grammar-token-depfile-pathname), which this is based on. There appear to be no documented escapes for `:` or `\n` in paths which might cause things to go sideways, but hopefully these are rare (`:` is banned on Windows for example).

Test added.